### PR TITLE
Execute jobs when in samples

### PR DESCRIPTION
**Pull request by alfred2g**
Execute jobs when not in CI

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -516,17 +535,6 @@ jobs:
...
-        uses: aws-actions/configure-aws-credentials@v1
-        with:
-          role-to-assume: ${{ env.CI_JOBS_ROLE }}
-          aws-region: ${{ env.AWS_DEFAULT_REGION }}
-      - name: run Jobs sample
-        run: |
-          python3 ${{ env.CI_UTILS_FOLDER }}/run_sample_ci.py --file ${{ env.CI_SAMPLES_CFG_FOLDER }}/ci_run_jobs_cfg.json
-      - name: run Mqtt5 Jobs sample
-        run: |
-          python3 ${{ env.CI_UTILS_FOLDER }}/run_sample_ci.py --file ${{ env.CI_SAMPLES_CFG_FOLDER }}/ci_run_jobs_mqtt5_cfg.json
```
[the remark this answers is not in the document]
Comment (alfred2g) [reply .github/workflows/ci.yml:529]: it is exactly like the servicetest one, I will add them back

File: servicetests/test_cases/test_jobs_execution.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+    thing_job = 'ERROR' 
+    i = 0;
+    while 'ERROR' in thing_job and i <= 3:
+        try:
+            job_id = secrets_client.get_secret_value(SecretId="ci/JobsServiceClientTest/job_id")["SecretString"]
+            thing_job = iot_client.describe_job_execution(jobId=job_id, thingName=thing_name)
+            print('thing job is {thing_job}');
+            if 'ERROR' in thing_job:
+               i = i + 1;
```
Comment (sfod) [inline servicetests/test_cases/test_jobs_execution.py:65]: Shouldn't we increment `i` in the Exception block also? Though, the limit should be increased.
Also, maybe it's worth adding a 1-second sleep after each unsuccessful attempt, to not bombard server with requests.

File: servicetests/test_cases/mqtt3_jobs_cfg.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            "data": "tests/JobsExecution/private.pem.key"
+        },
+        {
+            "name": "--thing_name",
+            "data": "ServiceTest_Jobs_$INPUT_UUID"
+        },
+        {
+            "name": "--job_id",
+            "data": "1"
+        }
```
[the remark this answers is not in the document]
Comment (alfred2g) [reply servicetests/test_cases/mqtt3_jobs_cfg.json:30]: no why?

File: servicetests/test_cases/mqtt5_jobs_cfg.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        {
+            "name": "--key",
+            "data": "tests/JobsExecution/private.pem.key"
+        },
+        {
+            "name": "--thing_name",
+            "data": "ServiceTest_Jobs_$INPUT_UUID"
+        },
+        {
+            "name": "--job_id",
```
[the remark this answers is not in the document]
Comment (alfred2g) [reply servicetests/test_cases/mqtt5_jobs_cfg.json:28]: no, why?

File: samples/jobs/mqtt5_job_execution/main.cpp [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+                fprintf(stderr, "Error %d occurred\n", ioErr);
+            }
+            publishDescribeJobExeCompletedPromise.set_value();
+        };
+
+        jobsClient.PublishDescribeJobExecution(
+            std::move(describeJobExecutionRequest), AWS_MQTT_QOS_AT_LEAST_ONCE, publishHandler);
+        publishDescribeJobExeCompletedPromise.get_future().wait();
+    }
+
```
[the remark this answers is not in the document]
Comment (alfred2g) [reply samples/jobs/mqtt5_job_execution/main.cpp:257]: copied it back here from the other PR

File: samples/jobs/job_execution/main.cpp
Comment (sfod) [inline samples/jobs/job_execution/main.cpp:1]: I got segfault for this sample. I didn't have any Job configured, maybe this case is not handled?

**Review by sfod — CHANGES_REQUESTED** [review]

File: samples/jobs/describe_job_execution/main.cpp
[the remark this answers is not in the document]
Comment (sfod) [reply samples/jobs/describe_job_execution/main.cpp:1]: I see that you removed this sample in the next PR, so nevermind

File: samples/jobs/mqtt5_job_execution/main.cpp [hunk 6]
```diff
@@ -167,16 +181,175 @@ int main(int argc, char *argv[])
...
+        }
+    };
+    auto subscribeHandler = [&](UpdateJobExecutionResponse *response, int ioErr) {
+        (void)response;
+        if (ioErr)
+        {
+            fprintf(stderr, "Error %d occurred\n", ioErr);
+            exit(-1);
+        }
+        fprintf(stdout, "Marked job %s currentJobId SUCCEEDED", currentJobId.c_str());
```
Comment (sfod) [inline samples/jobs/mqtt5_job_execution/main.cpp:323]:
```suggestion
        fprintf(stdout, "Marked job %s %s\n", currentJobId.c_str(), JobStatusMarshaller::ToString(status));
```

**Review by sfod — APPROVED** [review]

File: samples/utils/CommandLineUtils.cpp [hunk 7]
```diff
@@ -539,7 +541,7 @@ namespace Utils
         returnData.input_clientId =
             cmdUtils.GetCommandOrDefault(m_cmd_client_id, Aws::Crt::String("test-") + Aws::Crt::UUID().ToString());
         returnData.input_thingName = cmdUtils.GetCommandRequired(m_cmd_thing_name);
-        returnData.input_jobId = cmdUtils.GetCommandRequired(m_cmd_job_id);
+        returnData.input_jobId = cmdUtils.GetCommandOrDefault(m_cmd_job_id, "1");
```
Comment (sfod) [inline samples/utils/CommandLineUtils.cpp:544]: Debatable: Set to empty string by default. And in the samples request for a job description only if jobId is set.

**Pull request merged**